datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -167,14 +164,8 @@ impl ExecutionPlan for CoalesceBatchesExec {
 struct CoalesceBatchesStream {
     /// The input plan
     input: SendableRecordBatchStream,
-    /// The input schema
-    schema: SchemaRef,
-    /// Minimum number of rows for coalesces batches
-    target_batch_size: usize,
-    /// Buffered batches
-    buffer: Vec<RecordBatch>,
-    /// Buffered row count
-    buffered_rows: usize,
+    /// Buffer for combining batches
+    coalescer: BatchCoalescer,
```

Comment:
The whole point of the PR is to extract these fields and their management into a struct.